jupyterhub/spawner.py

```diff
@@ -723,6 +723,15 @@ def get_env(self):
             getattr(self.user, 'escaped_name', self.user.name),
             'activity',
         )
+        env['JUPYTERHUB_SPAWN_CALLBACK_URL'] = url_path_join(
+            self.hub.api_url,
+            'users',
+            # tolerate mocks defining only user.name
+            getattr(self.user, 'escaped_name', self.user.name),
```

Comment:
no need for getattr check here, since this is in the same codebase that defines escaped_name